sky/packages/sky/lib/src/rendering/paragraph.dart

```diff
@@ -15,59 +15,66 @@ export 'package:sky/src/painting/text_painter.dart';
 // differently. To work around this problem for now, we round fractional pixel
 // values up to the nearest whole pixel value. The right long-term fix is to do
 // layout using fixed precision arithmetic.
-double _applyFloatingPointHack(double layoutValue) {
+double applyFloatingPointHack(double layoutValue) {
   return layoutValue.ceilToDouble();
 }
 
 /// A render object that displays a paragraph of text
 class RenderParagraph extends RenderBox {
 
-  RenderParagraph(TextSpan text) : _textPainter = new TextPainter(text) {
+  RenderParagraph(
+      TextSpan text,
+      {allowLineWrap: true}
+  ) : textPainter = new TextPainter(text),
+      _allowLineWrap = allowLineWrap {
     assert(text != null);
   }
 
-  TextPainter _textPainter;
+  final TextPainter textPainter;
+
+  // Whether the text should be allowed to wrap to multiple lines.
+  final bool _allowLineWrap;
```

Comment:
Why is this an immutable field? I think it's more idiomatic to use a getter.
That is, have this here:

```dart
   bool get allowLineWrap => true;
```

...and then override it in the subclass to return false.